controllers/promiserelease_controller.go

```diff
@@ -160,9 +163,18 @@ func (r *PromiseReleaseReconciler) installPromise(o opts, promiseRelease *v1alph
 
 		return ctrl.SetControllerReference(promiseRelease, &existingPromise, r.Scheme)
 	})
-
 	if err != nil {
-		return nil
+		// Determine the reason for the failure to install the Promise
+		eventReason := "Failed to install Promise"
+		if errors.IsInvalid(err) {
+			eventReason = "Invalid Promise"
+		}
+
+		// Add an event to PromiseRelease about the failure of installing the Promise
+		r.EventRecorder.Eventf(promiseRelease, "Warning", eventReason,
+			"Failed to install Promise %q: %v", promise.GetName(), err)
```

Comment:
In case, eventReason holds the value as "Failed to install Promise" (i.e if errors.IsInvalid condition is false), won't the warning message have the same message twice?

Reply:
I think it is okay as they are 2 different fields. But I just changed the reason to `Failed`. I just found that one-word reasons are what the example shows here.

https://www.cncf.io/blog/2023/03/13/how-to-use-kubernetes-events-for-effective-alerting-and-monitoring/